caffe2/serialize/istream_adapter.cpp

```diff
@@ -0,0 +1,37 @@
+#include <caffe2/serialize/istream_adapter.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+
+IStreamAdapter::IStreamAdapter(std::istream* istream) : istream_(istream) {}
+
+size_t IStreamAdapter::size() {
+  auto prev_pos = istream_->tellg();
+  validate("getting the current position");
+  istream_->seekg(0, istream_->end);
+  validate("seeking to end");
+  auto result = istream_->tellg();
+  validate("getting size");
+  istream_->seekg(prev_pos);
+  validate("seeking to the original position");
+  return result;
+}
+
+size_t
+IStreamAdapter::read(uint64_t pos, void* buf, size_t n, const char* what) {
+  istream_->seekg(pos);
+  validate(std::string("seeking, ").append(what).c_str());
```

Comment:
How much do we care about efficiency of reading here? Each validate call is building a string and if we perform a lot of IStreamAdapter::read() calls, the validate would slow things down.

It seems like for large models this might be an issue. Do we need this function to take a `what` argument? Why don't we just have: `validate("seeking")` and `validate("reading")`

Reply:
it should be fine to just put `what` directly - I'm not even sure we need so detailed messages here